Cryfs: Update to 0.11.2 and dependency

jolan78 opened the pull request:
#### Description

Update Cryfs to 0.11.2 and the required range-v3 to 0.11.0

range-v3 has to be compiled without WERROR due to https://github.com/ericniebler/range-v3/issues/1685
is it OK to add a comment in the portfile describing this ?

I added python310 as a build dependency (used by the gitversion tool). python37 was previously used but has been deleted by @cjones051073 here: https://github.com/macports/macports-ports/commit/5c327283f4bb55253aab71f33f21e2418c236676#diff-5f109b2ba1617f8f933af9a477e0cec88b3fab6742a424a2d34a972801b25ec3L31. please comment if it is a mistake to add it back. I also patched the tool to explicitely use python3 instead of python, which the cryfs project did in https://github.com/cryfs/cryfs/commit/5f908c641cd5854b8a347f842b996bfe76a64577

###### Type(s)

- [ ] bugfix
- [X] enhancement
- [ ] security fix

###### Tested on

macOS 12.3.1 21E258 arm64
Xcode 13.3 13E113

###### Verification <!-- (delete not applicable items) -->
Have you

- [X] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [X] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [X] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? <!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [X] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [X] tested basic functionality of all binary files?

reneeotten commented on net/cryfs/files/gitversion_python3_fix.diff:
```diff
@@ -0,0 +1,11 @@
+ 
+ function (get_git_version OUTPUT_VARIABLE)
+-    EXECUTE_PROCESS(COMMAND python ${DIR_OF_GITVERSION_TOOL}/getversion.py
++    EXECUTE_PROCESS(COMMAND python3 ${DIR_OF_GITVERSION_TOOL}/getversion.py
```
the `python310` port does not install (or symlink) anything to `python3`, so you cannot rely on that. Instead please change `python` to, for example, `@@PYTHONBIN@@` and then do a reinplace in the `post-patch` phase to change that to `${prefix}bin/python3.10`

jolan78 commented:
Thank you @reneeotten for your comments, these are helpful.
I will fix these issues once I figure out a way to get range-v3 to activate properly: the include/module.modulemap file conflicts with ncurses and I am looking for a way to fix that properly (https://github.com/ericniebler/range-v3/issues/1522)

jolan78 commented:
Updated:
- macfuse's fuse.pc pointed to the incorrect /usr/local
- range-v3 headers have been moved to /opt/local/include/range-v3
- changed ref to python binary as suggested

reneeotten commented on net/cryfs/Portfile:
```diff
@@ -23,17 +24,24 @@ long_description          CryFS encrypts your files, so you can safely store the
 depends_lib-append        port:curl \
-                          port:libomp \
-                          port:osxfuse
+                          port:libomp
```
is this actually working with the default Apple clang compilers? Shouldn't this use something like `compiler.openmp_version ???` and such - see [here](https://trac.macports.org/wiki/CompilerSelection)?

jolan78 replied:
yes, cryfs detects OpenMP installed with macports (or homebrew): https://github.com/cryfs/cryfs/commit/08ed96680876c960d5aab0aabebb25206f0bda06#diff-1902f39b31e47c36e9d15ad9d1008e5b95fb0088f11f535bb8b1444a53c5d815
On my box it uses `/usr/bin/clang++` (Apple clang version 13.1.6 (clang-1316.0.21.2)) with `-Xpreprocessor -fopenmp -I/opt/local/include/libomp/ /opt/local/lib/libomp/libomp.dylib`
therefore the `compiler.openmp_version` option is not needed, unless I'm missing something ?

jolan78 replied:
I am marking this request as resolved. feel free to re-open it if I misunderstood.
